### PR TITLE
fix unencoded keys in load-base-split

### LINKED ISSUE #10542: load-base-split may split regions with unencoded keys
### What version of TiKV are you using?

[v5.0.0, v5.0.3], v5.1.0, master

### What operating system and CPU are you using?

Not important.

### Steps to reproduce

TiKV collects query statistics and key ranges for load-base-split, but the key range is raw key in `batch_get_command()` which should be memcomparable encoded. It results in TiKV splits regions with the key that TiDB can't recognize and TiDB can't access such a region until it's merged into its siblings.
https://github.com/tikv/tikv/blob/03044f20bd499f99ff2a556aaf78fc941019e95d/src/storage/mod.rs#L435-L442

### What did you expect?

TiDB works normally.

### What did happened?

TiDB reports
```
[2021/07/09 15:43:11.899 +08:00] [WARN] [backoff.go:331] ["pdRPC backoffer.maxSleep 40000ms is exceeded, errors:\nloadRegion from PD failed, key: \"mBootstra\\xffpKey\\x00\\x00\\x00\\x00\\xfb\\x00\\x00\\x00\\x00\\x00\\x00\\x00s\", err: invalid marker byte, group bytes \"t\\x80\\x00\\x00\\x00\\x00\\x00\\x00\\x0f\" at 2021-07-09T15:43:07.061309853+08:00\nloadRegion from PD failed, key: \"mBootstra\\xffpKey\\x00\\x00\\x00\\x00\\xfb\\x00\\x00\\x00\\x00\\x00\\x00\\x00s\", err: invalid marker byte, group bytes \"t\\x80\\x00\\x00\\x00\\x00\\x00\\x00\\x0f\" at 2021-07-09T15:43:09.483357169+08:00\nloadRegion from PD failed, key: \"mBootstra\\xffpKey\\x00\\x00\\x00\\x00\\xfb\\x00\\x00\\x00\\x00\\x00\\x00\\x00s\", err: invalid marker byte, group bytes \"t\\x80\\x00\\x00\\x00\\x00\\x00\\x00\\x0f\" at 2021-07-09T15:43:11.899411987+08:00"]
```

### PR DESCRIPTION
**Pull request by lhy1024**
Signed-off-by: lhy1024 <admin@liudos.us>

### What problem does this PR solve?

Issue Number: close #10542

### What is changed and how it works?

In batch_get_commands, there are different key format between others. This pr will fix it and add some test.

What's Changed:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Fix unencoded keys of `batch_get_command` in load-base-split
```

### CODE REVIEW

**Review by youjiali1995 — COMMENTED** [review]
The fix in storage/mod.rs LGTM.

File: src/storage/mod.rs [hunk 1]
```diff
@@ -1516,6 +1516,14 @@ impl<E: Engine, L: LockManager> Storage<E, L> {
...
                     };
                     let mut result = Vec::new();
+                    let mut key_ranges = vec![];
+                    for range in &ranges {
+                        key_ranges.push(build_key_range(
+                            &range.start_key,
+                            &range.end_key,
+                            reverse_scan,
+                        ));
+                    }
```
Comment (JmPotato) [inline src/storage/mod.rs:1526]: Why moving this part of the code to here?

Comment (lhy1024) [reply src/storage/mod.rs:1526]: `take_start_key` in Line1529 will replace it with empty.

Comment (JmPotato) [reply src/storage/mod.rs:1526]: I see, thx!

**Review by JmPotato — APPROVED** [review]

**Review by youjiali1995 — APPROVED** [review]

**Review by zhangjinpeng1987 — APPROVED** [review]
LGTM

**Pull request merged**